cmk/base/plugins/agent_based/cisco_asa_failover.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 tribe29 GmbH - License: GNU General Public License v2
+# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
+# conditions defined in the file COPYING, which is part of this source code package.
+
+#
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.4  'Failover LAN Interface'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.6  'Primary unit (this device)'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.7  'Secondary unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.4  2
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.6  9     < These two values flip during
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.7  10    < failover
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.4  'LAN_FO GigabitEthernet0/0.777'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.6  'Active unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.7  'Standby unit'
+
+
+# [['Failover LAN Interface', '2', 'LAN_FO GigabitEthernet0/0.777'],
+#  ['Primary unit', '9', 'Active unit'],
+#  ['Secondary unit (this device)', '10', 'Standby unit']]
+
+# failover off/ not configured
+# [['Failover LAN Interface', '3', 'not Configured'],
+#  ['Primary unit', '3', 'Failover Off'],
+#  ['Secondary unit (this device)', '3', 'Failover Off']]
+
+from dataclasses import dataclass
+from typing import List, Mapping, Any
+
+from cmk.base.plugins.agent_based.agent_based_api.v1.type_defs import (
+    DiscoveryResult,
+    CheckResult,
+    StringTable,
+)
+
+from cmk.base.plugins.agent_based.agent_based_api.v1 import (
+    register,
+    Service,
+    Result,
+    State,
+    SNMPTree,
+    startswith,
+    contains,
+    any_of,
+)
+
+
+def get_cisco_asa_state_name(st: str) -> str:
+    names = {
+        '1': 'other',
+        '2': 'up',
+        '3': 'down',
+        '4': 'error',
+        '5': 'overTemp',
+        '6': 'busy',
+        '7': 'noMedia',
+        '8': 'backup',
+        '9': 'active',
+        '10': 'standby',
+    }
+    return names.get(st, 'unknown %s' % st)
+
+
+@dataclass
+class Section:
+    failover: bool
+    local_role: str
+    local_status: str
+    local_status_detail: str
+    failver_link_status: str
+    remote_status: str
+
+
+def parse_cisco_asa_failover(string_table: List[StringTable]) -> Section:
+    failover = False
+    for role, status, detail in string_table[0]:
+        if 'this device' in role:
+            if not 'failover off' in detail.lower():
```

Comment:
failover = "failover off" not in detail.lower()

Reply:
removed this